src/com/dre/brewery/Barrel.java

```diff
@@ -551,27 +554,28 @@ public static void save(ConfigurationSection config, ConfigurationSection oldDat
 		}
 	}
 
-	public static class BarrelCheck extends BukkitRunnable {
+	public static class BarrelCheck extends UniversalRunnable {
 		@Override
 		public void run() {
 			boolean repeat = true;
 			while (repeat) {
 				if (check < barrels.size()) {
 					Barrel barrel = barrels.get(check);
 					if (!barrel.checked) {
-						Block broken = barrel.body.getBrokenBlock(false);
-						if (broken != null) {
-							P.p.debugLog("Barrel at "
-								+ broken.getWorld().getName() + "/" + broken.getX() + "/" + broken.getY() + "/" + broken.getZ()
-								+ " has been destroyed unexpectedly, contents will drop");
-							// remove the barrel if it was destroyed
-							barrel.remove(broken, null, true);
-						} else {
-							// Dont check this barrel again, its enough to check it once after every restart (and when randomly chosen)
-							// as now this is only the backup if we dont register the barrel breaking,
-							// for example when removing it with some world editor
-							barrel.checked = true;
-						}
+						P.getScheduler().runTaskTimer(barrel.getSpigot().getLocation(), () -> {
```

Comment:
As far as i understand the Scheduler, this will run the Task in the Region for that Location.
Considering we load all Barrels on server start, they could all over the world and in multiple worlds. Many of them not loaded and in regions that do not exist at the moment.
So this would cause Folia to create the missing regions if im correct?
Is there a way to only run the Task if the region exists?

Reply:
Could there be a case where a barrel would be in a non-existent region? If someone installed it, then the region was loaded before, right?

Reply:
Not if the plugin folder is moved to another server with a new map. If they forget about it, then it will throw errors. It may be an edgecase but important to have in mind